### ISSUE TITLE
[Deep Link/Custom Scheme] iOS not getting initial route from url after first launch

### ISSUE THREAD

**Issue by Oubi256** [lulupointu/vrouter#209]
Expected Behavior:
![image](https://user-images.githubusercontent.com/98897017/214243775-49f64c46-7ca6-4d92-89d4-2e0637423a57.png)

When cold starting IOS using deepLink, the initialUrl is not overwritten, instead, the initalUrl that was specified in the application is loaded first, and then the deepLink is followed. Instead, after a cold start, there should be a rewrite of the initalUrl so that the first page is immediately from deepLink, as is the case with a cold start on Android

Android no deeplink:
launchApp > nullPage > initalPage

deeplink Android cold:
launchApp > nullPage > ~initalPage~(rewrite) deeplinkPage

IOS customScheme cold:
launchApp > nullPage > **initalPage** > deeplinkPage

**Comment by lulupointu** [comment]
Can you provide a small reproducible example please ? 😊

**Comment by Oubi256** [comment]
> Can you provide a small reproducible example please ? 😊

Yes, here is an example and a link where you can test the launch of the desired page. On a cold start on this link, prev_url should always be null.
If you suddenly need to clearly show the difference between an android and an ios device, then write, I will record a video, I think it will not be difficult for me, but I hope this example will be enough.

https://github.com/Oubi256/vrouter_deeplink
![image](https://user-images.githubusercontent.com/98897017/214271950-885467d1-de1c-434f-b68c-a86b5d32efab.png)

**Comment by Oubi256** [comment]
@lulupointu Did you reproduced this issue? I was looking for a solution and came across a question regarding another router with the same problem, perhaps this is the flutter itself. But I think it might help you too.
https://github.com/flutter/flutter/issues/109165

**Comment by moshOntong-IT** [comment]
> @lulupointu Did you reproduced this issue? I was looking for a solution and came across a question regarding another router with the same problem, perhaps this is the flutter itself. But I think it might help you too. [flutter/flutter#109165](https://github.com/flutter/flutter/issues/109165)

does the VRouter supports deep link? is the issue still not solve?

**Comment by moshOntong-IT** [comment]
> > @lulupointu Did you reproduced this issue? I was looking for a solution and came across a question regarding another router with the same problem, perhaps this is the flutter itself. But I think it might help you too. [flutter/flutter#109165](https://github.com/flutter/flutter/issues/109165)
> 
> does the VRouter supports deep link? is the issue still not solve?

I am just a newbie here, I used Go router, but I am planning to find something router package because GoRouter nested mechanism not stable yet, and that so frustrate me.
